File: MAP/Lab 2/ToyInterpreter/src/view/commands/Command.java
package view.commands;

import controller.Controller;

public abstract class Command {
    protected Controller controller;
    private String key, description;

    public Command(String key, String description, Controller controller) {
        this.key = key;
        this.description = description;
        this.controller = controller;
    }

    public abstract void execute() throws Exception;
    public String getKey(){return key;}
    public String getDescription(){return description;}
}

File: MAP/Lab10/ConsoleApp1/ConsoleApp1/View/commands/Command.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ToyInterpreter.View.commands
{
    abstract class Command
    {
        protected Controller.Controller controller;
        protected String key, description;

        public Command(String key, String description, Controller.Controller controller)
        {
            this.key = key;
            this.description = description;
            this.Controller = controller;
        }

        public abstract void execute();

        protected Controller.Controller Controller { get => controller; set => controller = value; }

        public String getKey() { return key; }
        public String getDescription() { return description; }
    }
}
